synapse/cortex.py

```diff
@@ -5197,17 +5223,75 @@ async def count(self, text, opts=None):
 
         return i
 
+    async def _getMirrorOpts(self, opts):
+        mirropts = copy.deepcopy(opts)
+        mirropts['nomirror'] = True
+        mirropts['nexsoffs'] = (await self.getNexsIndx() - 1)
+        mirropts['nexstimeout'] = self.querymirroropts.get('timeout')
+        return mirropts
+
+    async def _getMirrorProxy(self):
+        proxy = None
+        try:
```

Comment:
Maybe we should check if there are any client connections and fall back immediately if not?

Reply:
We would need to check if our connection is a pool first, maybe Pool.proxy() should return immediately if no clients are configured?

Reply:
added a check for empty pools